src/ray/util/process.cc

```diff
@@ -133,15 +134,7 @@ class ProcessFD {
       int r = read(pipefds[0], &pid, sizeof(pid));
       (void)r;  // can't do much if this fails, so ignore return value
     }
-    if (decouple) {
-      fd = pipefds[0];  // grandchild, but we can use this to track its lifetime
-    } else {
-      fd = -1;  // direct child, so we can use the PID to track its lifetime
-      if (pipefds[0] != -1) {
-        close(pipefds[0]);
-        pipefds[0] = -1;
-      }
-    }
+    fd = pipefds[0];  // Use pipe to track process lifetime
```

Comment:
Can you expand on this comment? I assume the fd will close when the process dies?

Reply:
Yup exactly, I'll expand on it.